src/network/udpsocket.cpp

```diff
@@ -0,0 +1,92 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Socket for udp connections.
+ */
+
+#include "src/common/util.h"
+#include "src/common/memreadstream.h"
+#include "src/common/memwritestream.h"
+#include "src/common/error.h"
+
+#include "src/network/udpsocket.h"
+
+namespace Network {
+
+UDPSocket::UDPSocket(boost::asio::io_service &io) : _socket(io) {
+	_socket.open(boost::asio::ip::udp::v4());
+}
+
+UDPSocket::~UDPSocket() {
+	_socket.close();
+}
+
+void UDPSocket::bind(UDPEndpoint endpoint) {
+	_socket.bind(endpoint);
+}
+
+void UDPSocket::send(const Packet &packet) {
+	char message[packet.getSize()];
```

Comment:
While variable-length arrays do exist in C99, they do not exist in any current C++ standard. Last I heard, they're not even proposed for C++20.

You need to use a Common::ScopedArray<char> for that.